core/trino-main/src/main/java/io/trino/metadata/MetadataListing.java

```diff
@@ -83,83 +84,200 @@ public static SortedSet<String> listSchemas(Session session, Metadata metadata,
+                    .collect(toImmutableMap(Entry::getKey, Entry::getValue));
+        }
+        catch (RuntimeException exception) {
+            if (exception instanceof TrinoException trinoException) {
+                throw new TrinoException(
+                        trinoException::getErrorCode,
+                        "Error getting materialized views for catalog %s: %s".formatted(prefix.getCatalogName(), exception.getMessage()),
+                        exception);
+            }
+            throw new TrinoException(
+                    StandardErrorCode.GENERIC_INTERNAL_ERROR,
+                    "Error getting materialized views for catalog %s".formatted(prefix.getCatalogName()),
+                    exception);
+        }
     }
 
     public static Set<GrantInfo> listTablePrivileges(Session session, Metadata metadata, AccessControl accessControl, QualifiedTablePrefix prefix)
     {
-        List<GrantInfo> grants = metadata.listTablePrivileges(session, prefix);
-        Set<SchemaTableName> allowedTables = accessControl.filterTables(
-                session.toSecurityContext(),
-                prefix.getCatalogName(),
-                grants.stream().map(GrantInfo::getSchemaTableName).collect(toImmutableSet()));
+        try {
+            List<GrantInfo> grants = metadata.listTablePrivileges(session, prefix);
+            Set<SchemaTableName> allowedTables = accessControl.filterTables(
+                    session.toSecurityContext(),
+                    prefix.getCatalogName(),
+                    grants.stream().map(GrantInfo::getSchemaTableName).collect(toImmutableSet()));
 
-        return grants.stream()
-                .filter(grantInfo -> allowedTables.contains(grantInfo.getSchemaTableName()))
-                .collect(toImmutableSet());
+            return grants.stream()
+                    .filter(grantInfo -> allowedTables.contains(grantInfo.getSchemaTableName()))
+                    .collect(toImmutableSet());
+        }
+        catch (RuntimeException exception) {
+            if (exception instanceof TrinoException trinoException) {
+                throw new TrinoException(
+                        trinoException::getErrorCode,
+                        "Error listing table privileges for catalog %s: %s".formatted(prefix.getCatalogName(), exception.getMessage()),
+                        exception);
+            }
+            throw new TrinoException(
+                    StandardErrorCode.GENERIC_INTERNAL_ERROR,
+                    "Error listing table privileges for catalog %s".formatted(prefix.getCatalogName()),
+                    exception);
+        }
     }
 
     public static Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(Session session, Metadata metadata, AccessControl accessControl, QualifiedTablePrefix prefix)
+    {
+        try {
+            return doListTableColumns(session, metadata, accessControl, prefix);
```

Comment:
I like this approach of adding a `do` method so the original code is left alone. I think it's easier to read for both the code and the diff. Let's do that for all of them.